File: service/src/main/java/no/nav/melosys/service/dokument/brev/BrevDataAnmodningUnntak.java
package no.nav.melosys.service.dokument.brev;

import java.util.Set;

import no.nav.melosys.domain.VilkaarBegrunnelse;
import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet;
import no.nav.melosys.domain.kodeverk.yrker.Yrkesaktivitetstyper;

public class BrevDataAnmodningUnntak extends BrevData {
    public String arbeidsland;
    public AvklartVirksomhet hovedvirksomhet;
    public Yrkesaktivitetstyper yrkesaktivitet;
    public Set<VilkaarBegrunnelse> anmodningBegrunnelser;
    public Set<VilkaarBegrunnelse> anmodningUtenArt12Begrunnelser;

    public BrevDataAnmodningUnntak(String saksbehandler) {
        super(saksbehandler);
    }
}

File: service/src/main/kotlin/no/nav/melosys/service/dokument/brev/BrevDataAnmodningUnntak.kt
package no.nav.melosys.service.dokument.brev

import no.nav.melosys.domain.VilkaarBegrunnelse

import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet

import no.nav.melosys.domain.kodeverk.yrker.Yrkesaktivitetstyper

class BrevDataAnmodningUnntak(
    saksbehandler: String,
    val arbeidsland: String? = null,
    val hovedvirksomhet: AvklartVirksomhet? = null,
    val yrkesaktivitet: Yrkesaktivitetstyper? = null,
    var anmodningBegrunnelser: Set<VilkaarBegrunnelse> = emptySet(),
    val anmodningUtenArt12Begrunnelser: Set<VilkaarBegrunnelse> = emptySet(),
    val anmodningFritekst: String? = null
) : BrevData(saksbehandler = saksbehandler)
